File: src/gocode/util/search_utilities.go
package util

var commonSpecs []*SearchSpec = []*SearchSpec{
	&SearchSpec{"Google", "http://www.google.com/search?q="},
	&SearchSpec{"Yahoo", "http://search.yahoo.com/bin/search?p="},
	&SearchSpec{"Baidu", "http://www.baidu.com/baidu?word="},
}

func GetCommonSpecs() []*SearchSpec {
	return commonSpecs
}

func MakeURL(searchEngineName, searchString string) string {
	searchURL := ""
	for i := 0; i < len(commonSpecs); i++ {
		spec := commonSpecs[i]
		if spec.Name == searchEngineName {
			searchURL = spec.MakeURL(searchString)
			break
		}
	}
	return searchURL
}

File: src/util/SearchUtilities.java
package util;

public class SearchUtilities {
	private static SearchSpec[] commonSpecs = {
			new SearchSpec("Google", "http://www.google.com/search?q="),
			new SearchSpec("Yahoo", "http://search.yahoo.com/bin/search?p="),
			new SearchSpec("baidu", "http://www.baidu.com/baidu?word="),
	};
	
	public static SearchSpec[] getCommonSpecs() {
		return commonSpecs;
	}
	
	public static String makeURL(String searchEngineName, String searchString) {
		SearchSpec[] searchSpecs = getCommonSpecs();
		String searchURL = null;
		for (int i = 0; i < searchSpecs.length; i++) {
			SearchSpec spec = searchSpecs[i];
			if (spec.getName().equalsIgnoreCase(searchEngineName)) {
				searchURL = spec.makeURL(searchString);
				break;
			}
		}
		return searchURL;
	}
}
